### DISCUSSION TITLE
Mogrify vs Convert

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7159] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

What's the difference between mogrify and convert? Which one should I use?

**Comment by fmw42** [comment]
magick mogrify will process a whole folder of images, one at a time by itself.  Magick (don't use convert on IM 7) will process one image. Magick is more flexible and has more features.  But if you have a simple command and lots of images to process, then magick mogrify is more convenient.

**Reply by Jay123210599** [reply]
So magick and mogrify are the same thing?

**Comment by snibgo** [comment]
> So magick and mogrify are the same thing?

No.

`convert` is an old "legacy" program, provided for compatabilty only. Don't use it for new work. Instead, use `magick`. (Just `magick`, not `magick convert`.)

`magick` will process the files you name in the command. It will read and process all the files named. This might be multiple files, if you want to append them together into a single image, and similar operations.

`magick mogrify` or just `mogrify` has an inbuilt looping mechanism. You might specify that the input files are "*.jpg", etc. The program will read the first file and process it, then read the second file and process it, and so on. By default, it will overwrite each input file. This is convenient when you want to apply a fairly simple process to each of a set of files.

**Reply by fmw42** [reply]
With magick mogrify, you can also direct the output to a different (existing) directory using the -path argument.

Magick is also more flexible than magick mogrify, but usually just processes one input to create one output. However, you can have multiple inputs, if you want to combine them, but not multiple outputs at the same time as multiple inputs.
